scipy/interpolate/_aaa.py

```diff
@@ -0,0 +1,404 @@
+    errors : array
+        Error in the successive iterations of AAA.
+    
+    Methods
+    -------
+    poles()
+        Compute the poles of the rational approximation.
+    residues()
+        Compute the residues associated with the `poles` of the rational approximation.
+    roots()
+        Compute the roots(zeros) of the rational approximation.
+    
+
+    Warns
+    -----
+    RuntimeWarning
+        If `rtol` is not achieved in `max_terms` iterations.
+
+    See Also
+    --------
+    pade : Padé approximation
+
+    Notes
+    -----
+    At the :math:`m` th iteration (at which point there are :math:`m` terms), the
+    rational approximation in the AAA algorithm takes the barycentric form
+
+    .. math::
+
+        r(z) = n(z)/d(z) =
+        \frac{\sum_{j=1}^m\ w_j f_j / (z - z_j)}{\sum_{j=1}^m w_j / (z - z_j)},
+
+    where :math:`z_1,\dots,z_m` are real or complex support points selected from `z`,
+    :math:`f_1,\dots,f_m` are a set of real or complex data values, and
+    :math:`w_1,\dots,w_m` are real or complex weights. The algorithm then proceeds to
+    select the next support point :math:`z_{m+1}` is selected from the remaining
+    unselected points in `z` such that the nonlinear residual :math:`|f(z) - n(z)/d(z)|`
+    is maximised. The algorithm terminates when this maximum is less than
+    ``rtol * np.linalg.norm(f, ord=np.inf)``. This means the interpolation property is
+    only satisfied up to a tolerance. The weights are selected to solve the
+    least-squares problem
+
+    .. math::
+
+        \text{minimise}|fd - n| \quad \text{subject to} \quad
+        \sum_{i=1}^{m+1} w_i = 1,
+
+    over the unselected points in `z`.
+
+    References
+    ----------
+    .. [1] Y. Nakatsukasa, O. Sete, and L. N. Trefethen, "The AAA algorithm for
+            rational approximation", SIAM J. Sci. Comp. 40 (2018), A1494-A1522.
+            :doi:`10.1137/16M110612`
+
+    Examples
+    --------
+
+    Here we reproduce a number of the numerical examples from [1]_ as demonstration
+    of the functionality offered by this method.
+
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy.interpolate import AAA
+
+    For the first example we approximate the gamma function on ``[-3.5, 4.5]`` by
+    extrapolating from 100 samples in ``[-1.5, 1.5]``.
+
+    >>> from scipy.special import gamma
+    >>> sample_points = np.linspace(-1.5, 1.5, num=100)
+    >>> r = AAA(sample_points, gamma(sample_points))
+    >>> z = np.linspace(-3.5, 4.5, num=1000)
+    >>> fig, ax = plt.subplots()
+    >>> ax.plot(z, gamma(z), label="Gamma")
+    >>> ax.plot(sample_points, gamma(sample_points), label="Sample points")
+    >>> ax.plot(z, r(z).real, '--', label="AAA approximation")
+    >>> ax.set(xlabel="z", ylabel="r(z)", ylim=[-8, 8], xlim=[-3.5, 4.5])
+    >>> ax.legend()
+    >>> plt.show()
+
+    We can also view the poles of the rational approximation and their residue:
```

Comment:
```suggestion
    We can also view the poles of the rational approximation and their residues:
```
Just guessing based on the method name.